### PR TITLE
cli: Add `Request` `Response` `Headers` and `fetch` to `jsdom` environment

### PR DESCRIPTION
**Pull request by benjdlambert**

### CODE REVIEW

File: packages/cli/config/jest.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ function getRoleConfig(role) {
     case 'common-library':
     case 'frontend-plugin':
     case 'frontend-plugin-module':
-      return { testEnvironment: require.resolve('jest-environment-jsdom') };
+      return { testEnvironment: require.resolve('./jsdomEnvironment') };
```
Comment (Rugvip) [inline packages/cli/config/jest.js:51]: everything else in the config dir is namespaced, could we make this `jestJsdomEnvironment`?

**Review by Rugvip — COMMENTED** [review]
Sweet! This is indeed a nicer broader fix. Let's also clean up poliyfills in setupTests.ts

**Review by Rugvip — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]
🧹 :shipit:

**Review by Rugvip — CHANGES_REQUESTED** [review]
So we need to do a bit more thinking 😅

**Review by camilaibs — APPROVED** [review]
Approving discoverability changes 🙂 .

File: packages/cli/config/jest.js [hunk 2]
```diff
@@ -194,9 +194,16 @@ async function getProjectConfig(targetPath, extraConfig) {
     ...getRoleConfig(closestPkgJson?.backstage?.role),
   };
 
+  options.setupFilesAfterEnv = [];
+
+  if (options.testEnvironment === require.resolve('jest-environment-jsdom')) {
+    // FIXME https://github.com/jsdom/jsdom/issues/1724
+    options.setupFilesAfterEnv.push(require.resolve('cross-fetch/polyfill'));
```
Comment (Rugvip) [inline packages/cli/config/jest.js:201]: Feeling we want to make sure this is always first right? so `.unshift` given the change above

File: packages/cli/config/jest.js [hunk 3]
```diff
@@ -194,9 +194,16 @@ async function getProjectConfig(targetPath, extraConfig) {
     ...getRoleConfig(closestPkgJson?.backstage?.role),
   };
 
+  options.setupFilesAfterEnv = [];
```
Comment (Rugvip) [inline packages/cli/config/jest.js:197]: Feeling we should `options.setupFilesAfterEnv = options.setupFilesAfterEnv || []` here

**Review by Rugvip — CHANGES_REQUESTED** [review]
Yep let's do it, and then perhaps be able to just ship in in the env in 18

**Review by Rugvip — APPROVED** [review]
Sweet 😎 :shipit:

**Review by freben — APPROVED** [review]
Love it!

**Pull request merged**